File: server/src/main/java/message/scmessage/game/ScCharacterState.java
package message.scmessage.game;

import message.csmessage.game.CsCharacterState;
import message.scmessage.ScMessage;

public class ScCharacterState extends ScMessage {
    public String id;
    public float locationX;
    public float locationY;
    public float directionX;
    public float directionY;
    public int state;

    public ScCharacterState(CsCharacterState csCharacterState) {
        this.id = csCharacterState.id;
        this.locationX = csCharacterState.locationX;
        this.locationY = csCharacterState.locationY;
        this.directionX = csCharacterState.directionX;
        this.directionY = csCharacterState.directionY;
        this.state = csCharacterState.state;
    }
}

File: Client/Fruitster/Assets/Script/NetConnection/Messages/ScMessages/GamePlay/ScCharacterState.cs
using System;
using Script.Character;
using UnityEngine;

namespace Script.Messages.ScMessages
{
    [Serializable]
    public class ScCharacterState : ScMessage
    {
        public string id;
        public float locationX;
        public float locationY;
        public float directionX;
        public float directionY;
        public ObjectState state;
        public override void OnMessage(Session session)
        {
            if (id != UserProperties.MainPlayer.userID)
            {
                GameObject go = PlaySceneScript.Instance.GetPlayerById(id);
                ObjectUpdateMoving objectUpdateMoving = go.GetComponent<ObjectUpdateMoving>();
                if (objectUpdateMoving != null)
                {
                    objectUpdateMoving.location = new Vector3(locationX, locationY,objectUpdateMoving.location.z);
                    objectUpdateMoving.direction = new Vector3(directionX, directionY,objectUpdateMoving.location.z);
                    go.GetComponent<ObjectScript>().state = state;
                }

            } 
        }
    }
}